app/models/mixins/aggregation_mixin.rb

```diff
@@ -1,23 +1,22 @@
 module AggregationMixin
   extend ActiveSupport::Concern
   included do
-    virtual_column :aggregate_cpu_speed,       :type => :integer, :uses => :hosts
-    virtual_column :aggregate_cpu_total_cores, :type => :integer, :uses => :hosts
-    virtual_column :aggregate_physical_cpus,   :type => :integer, :uses => :hosts
-    virtual_column :aggregate_memory,          :type => :integer, :uses => :hosts
-    virtual_column :aggregate_vm_cpus,         :type => :integer, :uses => :vms_and_templates
-    virtual_column :aggregate_vm_memory,       :type => :integer, :uses => :vms_and_templates
-    virtual_column :aggregate_disk_capacity,   :type => :integer, :uses => :hosts
+    virtual_aggregate :aggregate_vm_memory, :vm_hardwares, :sum, :memory_mb
+    virtual_aggregate :aggregate_vm_cpus, :vm_hardwares, :sum, :cpu_sockets
+    virtual_aggregate :aggregate_physical_cpus, :host_hardwares, :sum, :cpu_sockets
+    virtual_aggregate :aggregate_disk_capacity, :host_hardwares, :sum, :disk_capacity
+    virtual_aggregate :aggregate_memory, :host_hardwares, :sum, :memory_mb
+    virtual_aggregate :aggregate_cpu_total_cores, :host_hardwares, :sum, :cpu_total_cores
+    virtual_aggregate :aggregate_cpu_speed, :host_hardwares, :sum, :aggregate_cpu_speed
```

Comment:
Pedantic:  Can we alphabetize these calls like they were (almost) before?

A)  It would be easier to review and compare with what was there previously
B)  I am anal retentive.

Reply:
Actually, with such a request, I will do it for you so it is just a copy past exercise...

```ruby
virtual_aggregate :aggregate_cpu_speed,       :host_hardwares, :sum, :aggregate_cpu_speed
virtual_aggregate :aggregate_cpu_total_cores, :host_hardwares, :sum, :cpu_total_cores
virtual_aggregate :aggregate_disk_capacity,   :host_hardwares, :sum, :disk_capacity
virtual_aggregate :aggregate_memory,          :host_hardwares, :sum, :memory_mb
virtual_aggregate :aggregate_physical_cpus,   :host_hardwares, :sum, :cpu_sockets
virtual_aggregate :aggregate_vm_cpus,         :vm_hardwares,   :sum, :cpu_sockets
virtual_aggregate :aggregate_vm_memory,       :vm_hardwares,   :sum, :memory_mb
```

Reply:
Yes I formatted the spacing of it too... did I mention I am anal retentive?